src/hash.h

```diff
@@ -179,6 +180,9 @@ uint256 SerializeHash(const T &obj, int nType = SER_GETHASH,
     return ss.GetHash();
 }
 
+/** Single-SHA256 a 32-byte input (represented as uint256). */
+NODISCARD uint256 SHA256Uint256(const uint256& input);
```

Comment:
What is this? All I can find is this stuff https://riptutorial.com/cplusplus/example/19006/--nodiscard--

Reply:
In attributes.h: `#define NODISCARD __attribute__((warn_unused_result))`

It's basically the same as `#[must_use]` in Rust.